app/models/folder.rb

```diff
@@ -48,11 +50,17 @@ def size
     ""
   end
 
-  def display_privacy
-    ""
-  end
+  def self.public_top_folders
+    query = "SELECT folders.id
+      FROM folders
+      INNER JOIN folders as parents
+              ON folders.parent_id = parents.id
+      WHERE folders.is_private = false
+        AND parents.is_private = true
+      ;"
+    results = ActiveRecord::Base.connection.execute(query)
```

Comment:
I couldn't figure out how to do with query with ActiveRecord since it needs to join with itself.... If anyone can figure it out, I'll give you super special Ashley Brownie Points!